### PR TITLE
perf: set appropiate/missing lazy loading events

### PR DESCRIPTION
**Pull request by manuuurino**
## 📑 Description

Noticed some lazy loading improvements from my personal config. These were the one i am using, sorry but i cannot check all plugins, i hope i can improve the startup/running performance a tiny bit further.

### CODE REVIEW

File: lua/astrocommunity/pack/yaml/init.lua [hunk 1]
```diff
@@ -1,7 +1,8 @@
 return {
   {
     "b0o/SchemaStore.nvim",
-    lazy = true,
+    event = "User AstroFile",
+    ft = { "yaml", "json" },
```
Comment (Uzaaft) [inline lua/astrocommunity/pack/yaml/init.lua:5]: Shouldn't toml also be added?

Comment (Uzaaft) [reply lua/astrocommunity/pack/yaml/init.lua:5]: My bad, taplo has it built in by default.

File: lua/astrocommunity/editing-support/true-zen-nvim/init.lua [hunk 2]
```diff
@@ -1,5 +1,6 @@
 return {
   "Pocco81/true-zen.nvim",
+  event = "VeryLazy",
```
Comment (mehalter) [inline lua/astrocommunity/editing-support/true-zen-nvim/init.lua:3]:
```suggestion
```



This should not be lazy loaded on event. It is lazy loaded on module and keybinding

File: lua/astrocommunity/editing-support/vim-move/init.lua [hunk 3]
```diff
@@ -1,4 +1,4 @@
 return {
   "matze/vim-move",
-  event = "BufEnter",
+  event = "User AstroFile",
```
Comment (mehalter) [inline lua/astrocommunity/editing-support/vim-move/init.lua:3]:
```suggestion
  event = { "User AstroFile", "InsertEnter" },
```

This lets it work on new files as well

File: lua/astrocommunity/editing-support/true-zen-nvim/init.lua
Comment (luxus) [inline lua/astrocommunity/editing-support/true-zen-nvim/init.lua:1]: will it not get loaded when using the mappings?

File: lua/astrocommunity/lsp/lsp-inlayhints-nvim/init.lua [hunk 4]
```diff
@@ -1,6 +1,7 @@
 return {
   {
     "lvimuser/lsp-inlayhints.nvim",
+    event = "User AstroLspSetup",
```
Comment (mehalter) [inline lua/astrocommunity/lsp/lsp-inlayhints-nvim/init.lua:4]:
```suggestion
```

This lazy loading is handled by the autocmd and should not be loaded on event. This is less performant

File: lua/astrocommunity/pack/yaml/init.lua (hunk 1 above)
Comment (mehalter) [inline lua/astrocommunity/pack/yaml/init.lua:5]:
```suggestion
    lazy = true,
```

this should be reverted. This is more performant as is. It is lazy loaded based on module only when needed.

**Review by mehalter — CHANGES_REQUESTED** [review]
A few of these need to be reverted because they are significantly less performant. Other than that, thanks for the recommendations!

**Review by mehalter — APPROVED** [review]
I went in and resolved the performance issues. Thanks!

**Pull request merged**